modules/reindex/build.gradle

```
@@ -67,6 +67,7 @@ dependencies {
   api project(":libs:opensearch-ssl-config")
   // for http - testing reindex from remote
   testImplementation project(':modules:transport-netty4')
+  testImplementation "org.bouncycastle:bctls-fips:${versions.bouncycastle_tls}"
```

Since we have `fips` configuration:

```suggestion
  testFipsImplementation "org.bouncycastle:bctls-fips:${versions.bouncycastle_tls}"
```

---

From my understanding this will have the same effect as leaving it as is: `testImplementation` declared dependencies won't propagate transitively - e.g. BCTLS won't propagate to `:modules:opensearch-dashboards`.

---

> From my understanding this will have the same effect as leaving it as is: `testImplementation` declared dependencies won't propagate transitively - e.g. BCTLS won't propagate to `:modules:opensearch-dashboards`.

Correct, the suggestion here is to keep fips related dependencies under fips configuration - no visible impact on the dependencies propagation but consistency of dependency management

---

Certainly! Let's name it `testFipsOnly` then and do you want BC deps inside `:test:framework` `:distribution:tools:plugin-cli` `:client:test` to be changed accordingly?

---

That would be great I think, thank you